src/Microsoft.IdentityModel.Logging/IdentityModelTelemetryUtil.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Reflection;
+using System.Net.Http;
+
+namespace Microsoft.IdentityModel.Logging
```

Comment:
Not sure if this the best place for this class to live.

Reply:
Perhaps it's better to move it under Tokens. It would eliminate the need for System.Http.Net dependency and some internalsVisibleTo in this project.

Reply:
+1

Reply:
M.IM.Tokens also does not have System.Http.Net dependency and M.IM.Protocols(which has HttpDocumentRetreiver) is not set as a friend assembly to it.

Reply:
@RojaEnnam i think logging is the right assembly.
To solve the dependency on System.Net, SetTelemetry would need to be replaced with a method that returns the metadata as a IDictionary or some other structure. Then, each user would need to added the values to the HttpRequestMessage.
@GeoK  I don't see anyone way to take System.Net.Http out of the picture.